src/reconcile/rec_visibility.c

```diff
@@ -277,6 +279,18 @@ __rec_need_save_upd(
     if (F_ISSET(r, WT_REC_CHECKPOINT) && upd_select->upd == NULL)
         return (false);
 
+    /*
+     * No need to save the update chain when it meets the following:
+     * 1. No need to restore the update chain
+     * 2. No on-disk entry exist
+     * 3. No further updates exist in the update chain to be written to the history store.
+     */
+    supd_restore =
+      F_ISSET(r, WT_REC_EVICT) && (has_newer_updates || F_ISSET(S2C(session), WT_CONN_IN_MEMORY));
+
+    if (!supd_restore && vpack == NULL && upd_select->upd != NULL && upd_select->upd->next == NULL)
```

Comment:
Can we walk to next update that us not aborted?

Reply:
Yes, added a loop to check for a valid update before saving the update chain.